### PR TITLE
Added context managers for automatically closing fds

### PR DESCRIPTION
**Pull request by millefalcon**
### Issue Reference
This PR address the issue to with forgetting to close file descriptors.

### Summarize
Used `with` context manager for cleaner and better code.

### CODE REVIEW

File: mainapp/admin.py [hunk 1]
```diff
@@ -49,47 +48,49 @@ class VolunteerAdmin(admin.ModelAdmin):
...
-            s = list(s)
-            s[6] = mapper[s[6]]
-            writer.writerow(s)
-        f.close()
-        f = open('vol.csv', 'r')
-        response = HttpResponse(f, content_type='text/csv')
-        response['Content-Disposition'] = 'attachment; filename=Volunteers.csv'
+        with open('vol.csv', 'w') as file_:
+            writer = csv.writer(file_)
+            l = []
```
Comment (arunk-iq) [inline mainapp/admin.py]: Could have used a list comprehension here and better not to use single lettered variable name

Comment (arunk-iq) [reply mainapp/admin.py]: ``volunteer_list = [vol.name for vol in Volunteer._meta.get_fields()]``
``writer.writerow(volunteer_list)``

The similar usages repeats in below code. They are better than loops.

File: mainapp/admin.py [hunk 2]
```diff
@@ -49,47 +48,49 @@ class VolunteerAdmin(admin.ModelAdmin):
     readonly_fields = ('joined',)
     list_display = ('name', 'phone', 'organisation', 'joined')
     list_filter = ('district', 'joined',)
+
     def download_csv(self, request, queryset):
```
Comment (bhumilharia) [inline mainapp/admin.py]: Shouldn't we be caching/memoizing the result for this? This probably adds to the performance/uptime issues.
How often is this hit?

**Comment by han-ish** [comment]
I could have used comprehensions, but was only going for where it was
crutial.
@bhumilharia Yeah, need to find out how it gets hit, and some form of
caching/memoization should be done.

**Pull request closed without merging**